### PR TITLE
Fix sample config

### LINKED ISSUE #14840: Uncommented config options in galaxy.yml
In Galaxy config, `galaxy.yml.sample`, the following should be commented out by default. Otherwise, if any other config option is uncommented local Galaxy instance does not start. Fix has been suggested by @davelopez 
 
```
task_routes:
-      galaxy.fetch_data: galaxy.external
-      galaxy.set_job_metadata: galaxy.external
```

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: latest dev branch
Commit: 

**To Reproduce**
1. Take latest from Galaxy dev
2. Uncomment any config option in galaxy.yml
3. Start local Galaxy instance
4. See the following error: ``` galaxy/config/galaxy.yml", line 21, column 1 expected <block end>, but found '<block mapping start>' ```

### PR DESCRIPTION
**Pull request by davelopez**
Fixes #14840 

Auto comment out complex default values.

## How to test the changes?
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [x] Instructions for manual testing are as follows:
  Follow the steps on #14840 

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by dannon** [comment]
Isn't this going to just get regenerated the other way without changing the schema?

**Comment by davelopez** [comment]
Good point, I didn't realize it was auto-generated :rofl: 
Thanks!

**Comment by jdavcs** [comment]
The problem is that the yaml file is generated incorrectly. The function at fault is in `lib/galaxy/config/config_manage.py`: the whole `celery_conf` block should be commented out by default - like everything else.

**Comment by nsoranzo** [comment]
For context, the regeneration is usually done with: `make config-rebuild` .

`lib/galaxy/config/config_manage.py` hadn't had to deal with YAML as default values before (only str, int, float and bool), so it's not surprising it doesn't work.

**Comment by davelopez** [comment]
Thank you all! I would have spent much more time figuring out how it was generated.
I think it works now with this little hack :)

**Review by jdavcs — APPROVED** [review]
Looks good! Simple :)

File: lib/galaxy/config/config_manage.py [hunk 1]
```diff
@@ -511,6 +511,7 @@ def _write_option(args, f, key, option_value, as_comment=False):
         comment += "\n".join("\n".join(YAML_COMMENT_WRAPPER.wrap(_)) for _ in desc.split("\n\n")) + "\n"
     as_comment_str = "#" if as_comment else ""
     key_val_str = yaml.dump({key: value}, width=math.inf).lstrip("{").rstrip("\n}")
+    key_val_str = key_val_str.replace("\n", f"\n{as_comment_str}")
```
Comment (nsoranzo) [inline lib/galaxy/config/config_manage.py]: Maybe add an explaining comment? The code in this file is often quite difficult to understand.

File: lib/galaxy/config/config_manage.py [hunk 2]
```diff
@@ -511,6 +511,7 @@ def _write_option(args, f, key, option_value, as_comment=False):
...
     lines = f"{comment}{as_comment_str}{key_val_str}"
     lines_idented = "\n".join(f"  {line}" for line in lines.split("\n"))
     f.write(f"{lines_idented}\n\n")
```
Comment (nsoranzo) [inline lib/galaxy/config/config_manage.py]: I think this is more readable:

```suggestion
    key_val_str = yaml.dump({key: value}, width=math.inf).lstrip("{").rstrip("\n}")
    if as_comment:
        # key_val_str can be span multiple lines, e.g. if value is a dict
        key_val_str = "\n".join(f"#{row}" for row in key_val_str.split("\n"))
    lines = f"{comment}{key_val_str}"
    lines_indented = "\n".join(f"  {line}" for line in lines.split("\n"))
    f.write(f"{lines_indented}\n\n")
```

**Pull request merged**